### PR TITLE
Site Indictator: Remove calls to `recordGoogleEvent()`

### PR DESCRIPTION
**Pull request by danielbachhuber**
Fixes https://github.com/Automattic/dotcom-forge/issues/3490

## Proposed Changes

Removes calls to `recordGoogleEvent()` from `<SiteIndicator />` and `<Site />`:

CleanShot 2023-12-12 at 03 50 19@2x

Google Analytics events are no longer in use: p1693487681855679-slack-C03N25JPCE4

## Testing Instructions

1. Interact with a `! isPreviewable` site (e.g. P2) and verify `calypso_mysites_sidebar_view_site_unpreviewable_clicked` still fires as expected.
2. Interact with a site that has a plugin update available and verify the link to `/wp-admin/update-core.php` still works as expected.
3. Interact with a site that has a broken Jetpack connection and verify `calypso_jetpack_site_indicator_disconnect_start` still fires when you start a disconnect.

### CODE REVIEW

File: client/my-sites/site-indicator/index.jsx [hunk 1]
```diff
@@ -146,10 +135,9 @@ export class SiteIndicator extends Component {
 		);
 	}
 
-	recordEvent( event, total ) {
+	recordEvent() {
 		window.scrollTo( 0, 0 );
 		this.setState( { expand: false } );
```
Comment (sejas) [inline client/my-sites/site-indicator/index.jsx]: If we remove these parameters (`event`, `total`).
Should we remove them also from the calls?

https://github.com/Automattic/wp-calypso/blob/e4f9ed11816d51c78d2e4e42c138909e3928e4fd/client/my-sites/site-indicator/index.jsx#L144-L170

**Review by sejas — COMMENTED** [review]
It looks good, I added a comment suggesting to remove more code when we call `this.recordEvent`.

File: client/my-sites/site-indicator/index.jsx (hunk 1 above)
Comment (danielbachhuber) [reply client/my-sites/site-indicator/index.jsx]: @sejas Good catch. I did some refactoring with 7313ed11b5901c91a71676638ca3bfb70ec5bd12

Comment (sejas) [reply client/my-sites/site-indicator/index.jsx]: Great refactor!! ✨

**Pull request merged**